### PR TITLE
libc/picolibc: Split hooks into separate files

### PR DESCRIPTION
**Pull request by keith-packard**
This splits the picolibc helper functions into separate files instead of smashing them all together. No functional changes are intended, this is just simple cleanup.

### CODE REVIEW

**Comment by rruuaanng** [comment]
I'm wondering if it's necessary.

**Comment by keith-packard** [comment]
> I'm wondering if it's necessary.

Not really? It makes the code a bit easier to manage as the files are shorter and have more obvious content.

**Review by stephanosio — APPROVED** [review]

File: lib/libc/picolibc/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,15 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
-zephyr_library_sources(libc-hooks.c)
-
-# Do not allow LTO when compiling libc-hooks.c file
-set_source_files_properties(libc-hooks.c PROPERTIES COMPILE_OPTIONS $<TARGET_PROPERTY:compiler,prohibit_lto>)
```
Comment (aescolar) [inline lib/libc/picolibc/CMakeLists.txt:7]: This is not needed anymore?

**Review by rruuaanng — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

File: lib/libc/picolibc/CMakeLists.txt (hunk 1 above)
Comment (aescolar) [reply lib/libc/picolibc/CMakeLists.txt:7]: Fair enough. From the commit message it kind of seemed like it may have been an oversight to remove this :)

Comment (keith-packard) [reply lib/libc/picolibc/CMakeLists.txt:7]: yeah, I hadn't remembered that part of the change as I did this a long time ago. Good to review and make sure nothing got missed.

**Review by aescolar — APPROVED** [review]

**Pull request merged**